sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt

```diff
@@ -0,0 +1,627 @@
+                  .styleURI(STYLE)
+                  .minZoom(ZOOM.toInt().toByte())
+                  .maxZoom(ZOOM.toInt().toByte())
+                  .build()
+              )
+            )
+          )
+          .metadata(Value(TILE_REGION_METADATA))
+          .tileLoadOptions(
+            TileLoadOptions.Builder()
+              .criticalPriority(false)
+              .acceptExpired(true)
+              .networkRestriction(NetworkRestriction.NONE)
+              .build()
+          )
+          .build(),
+        {
+          tileRegionProgress = it
+          cancelableTileStoreTask?.cancel()
+        }
+      ) {
+        if (it.isValue) {
+          it.value?.let { region ->
+            tileRegion = region
+          }
+        }
+        if (it.isError) {
+          Logger.e(TAG, "Error: ${it.error}")
+          it.error?.let { error ->
+            tileRegionError = error
+          }
+        }
+        latch.countDown()
+      }
+    }
+    if (!latch.await(30, TimeUnit.SECONDS)) {
+      throw TimeoutException()
+    } else {
+      Assert.assertNull(tileRegion)
+      Assert.assertNotNull(tileRegionError)
+      Assert.assertNotNull(tileRegionProgress)
+
+      Assert.assertNotEquals(
+        tileRegionProgress?.requiredResourceCount,
+        tileRegionProgress?.completedResourceCount
+      )
+      Assert.assertEquals(TileRegionErrorType.CANCELED, tileRegionError?.type)
+    }
+  }
+
+  // TODO revisit after https://github.com/mapbox/mapbox-sdk-common/issues/1311
+  // this test must behave same as resourceLoadingFromTileStoreMapboxSwitch
+  // but for now we could check for style loaded event only to style is loaded
+  @Test
+  fun resourceLoadingFromTileStoreAirplaneMode() {
+    disableAmbientCache()
+    loadTileStoreWithStylePack()
+    val latch = CountDownLatch(2)
+    var resourceRequests = 0
+    var mapLoadingErrorCount = 0
+    val observer = object : Observer() {
+      override fun notify(event: Event) {
+        Logger.e(TAG, "type ${event.type}, data ${event.data.toJson()}")
+        val data = event.getResourceEventData()
+        if (!data.cancelled && data.dataSource == "database" && data.request.kind == "tile") {
+          resourceRequests++
+        }
+        if (event.type == MapEvents.MAP_LOADING_ERROR) {
+          mapLoadingErrorCount++
+        }
+        if (event.type == MapEvents.MAP_LOADED || event.type == MapEvents.STYLE_LOADED) {
+          latch.countDown()
+        }
+      }
+    }
+    switchAirplaneMode()
+    prepareMapView(
+      observer,
+      listOf(
+        MapEvents.RESOURCE_REQUEST,
+        MapEvents.MAP_LOADED,
+        MapEvents.STYLE_LOADED,
+        MapEvents.MAP_LOADING_ERROR
+      )
+    )
+    try {
+      // we do not throw exception here but verify request count
+      latch.await(10, TimeUnit.SECONDS)
+      // verify resource requests came from tile store
+      MatcherAssert.assertThat(resourceRequests, Matchers.greaterThan(0))
```

Comment:
shall we also assert that the MAP_LOADED and STYLE_LOADED event are actually emitted?

Reply:
Full functionality of this test is broken as stated in a bunch of TODOs - so let's leave as is now to verify that tiles are loaded.

Reply:
let's ticket out the tailwork in this repo to improve these test when up steam is fixed.

Reply:
https://github.com/mapbox/mapbox-maps-android/issues/297